natter-api/src/main/java/com/manning/apisecurityinaction/controller/SpaceController.java

```diff
@@ -50,85 +56,120 @@ public JSONObject createSpace(Request request, Response response) {
   public JSONObject postMessage(Request request, Response response) {
     var spaceId = Long.parseLong(request.params(":spaceId"));
     var json = new JSONObject(request.body());
+
     var user = json.getString("author");
+
     if (!user.matches("[a-zA-Z][a-zA-Z0-9]{0,29}")) {
+
       throw new IllegalArgumentException("invalid username");
+
     }
     var message = json.getString("message");
+
     if (message.length() > 1024) {
+
       throw new IllegalArgumentException("message is too long");
+
     }
 
+
     return database.withTransaction(tx -> {
+
       var msgId = database.findUniqueLong(
           "SELECT NEXT VALUE FOR msg_id_seq;");
+
       database.updateUnique(
+
           "INSERT INTO messages(space_id, msg_id, msg_time," +
               "author, msg_text) " +
               "VALUES(?, ?, current_timestamp, ?, ?)",
+
           spaceId, msgId, user, message);
 
       response.status(201);
+
       var uri = "/spaces/" + spaceId + "/messages/" + msgId;
+
       response.header("Location", uri);
+
       return new JSONObject().put("uri", uri);
+
     });
   }
 
   public Message readMessage(Request request, Response response) {
     var spaceId = Long.parseLong(request.params(":spaceId"));
-    var msgId = Long.parseLong(request.params(":msgId"));
 
+    var msgId = Long.parseLong(request.params(":msgId"));
     var message = database.findUnique(Message.class,
+
         "SELECT space_id, msg_id, author, msg_time, msg_text " +
             "FROM messages WHERE msg_id = ? AND space_id = ?",
         msgId, spaceId);
-
     response.status(200);
     return message;
   }
 
   public JSONArray findMessages(Request request, Response response) {
+
+
     var since = Instant.now().minus(1, ChronoUnit.DAYS);
+
     if (request.queryParams("since") != null) {
+
       since = Instant.parse(request.queryParams("since"));
     }
+
     var spaceId = Long.parseLong(request.params(":spaceId"));
 
     var messages = database.findAll(Long.class,
         "SELECT msg_id FROM messages " +
             "WHERE space_id = ? AND msg_time >= ?;",
+
         spaceId, since);
 
+
     response.status(200);
     return new JSONArray(messages.stream()
+
         .map(msgId -> "/spaces/" + spaceId + "/messages/" + msgId)
+
         .collect(Collectors.toList()));
   }
 
   public static class Message {
+
     private final long spaceId;
     private final long msgId;
+
     private final String author;
     private final Instant time;
     private final String message;
 
     public Message(long spaceId, long msgId, String author,
+
         Instant time, String message) {
+
       this.spaceId = spaceId;
       this.msgId = msgId;
       this.author = author;
       this.time = time;
+
       this.message = message;
     }
     @Override
+
+
     public String toString() {
```

Comment:
1. Override function is not used anywhere
2. General functions should be added in a central class to be used for all controllers and classes